src/GovUk.Education.ExploreEducationStatistics.Data.Api/Services/Cache/ReleaseSubjectsCacheKey.cs

```diff
@@ -0,0 +1,35 @@
+﻿#nullable enable
+using System;
+using GovUk.Education.ExploreEducationStatistics.Common;
+using GovUk.Education.ExploreEducationStatistics.Common.Cache.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using static GovUk.Education.ExploreEducationStatistics.Common.Services.FileStoragePathUtils;
+
+namespace GovUk.Education.ExploreEducationStatistics.Data.Api.Services.Cache
+{
+    public record ReleaseSubjectsCacheKey : IBlobCacheKey
+    {
+        private string PublicationSlug { get; }
+        private string ReleaseSlug { get; }
+        
+        public Guid ReleaseId { get; }
```

Comment:
It seems strange that `ReleaseId` has no bearing here on what is cached, i.e. it's not used in the 'key', but I think it's because you want to use it to transfer the release id between methods taking this cache key as a param?

Reply:
Exactly.  I think there's a few things we'll probably want to be able to do in order to make the generation of cache keys a bit smarter, so that a method can just take a Release Id, generate a cache key from it, and cache the method's results in one go, thereby removing the need to do this kind of workaround, but I think that's a job for another day.